modules/indexer/code/elastic_search.go

```diff
@@ -449,4 +480,33 @@ func (b *ElasticSearchIndexer) Search(repoIDs []int64, language, keyword string,
 }
 
 // Close implements indexer
-func (b *ElasticSearchIndexer) Close() {}
+func (b *ElasticSearchIndexer) Close() {
+	close(b.stopTimer)
```

Comment:
Almost always safer to wrap `close(...)` in a `select` as this prevents `panic`s

```suggestion
	select {
		case <-b.stopTimer:
		default:
			close(b.stopTimer)
	}
```

Did you consider using a context, cancel pair? Not saying that this would be the right answer - context, cancel works best if you can create a hierarchy of cancellation - but if you need things to happen in certain orders then cancels may not be best.